src/extension.ts

```diff
@@ -6,32 +6,33 @@
  **/
 import * as vscode from 'vscode';
 import * as commands from './commands';
-import { sync } from 'cross-spawn';
 import { getTestOutlineProvider } from './views/testOutlineProvider';
 import { AgentTestRunner } from './views/testRunner';
 import { Commands } from './enums/commands';
 import type { AgentTestGroupNode, TestNode } from './types';
 import { CoreExtensionService } from './services/coreExtensionService';
+
 // This method is called when your extension is activated
-// Your extension is activated the very first time the command is executed
-// see "contributes" property in package.json for command list
 export async function activate(context: vscode.ExtensionContext) {
   const extensionHRStart = process.hrtime();
 
   try {
-    // We need to do this first in case any other services need access to those provided by the core extension
-    CoreExtensionService.loadDependencies(context);
+    // Load dependencies in the background to avoid blocking activation
+    CoreExtensionService.loadDependencies(context).catch((err: Error) =>
+      console.error('Error loading core dependencies:', err)
+    );
 
-    const versions = sync('sf', ['version', '--verbose', '--json'], { shell: true, encoding: 'utf8' });
-    if (!versions?.output?.toString().includes('agent')) {
-      throw new Error('sf CLI + plugin-agent installed required');
-    }
-    const disposables: vscode.Disposable[] = [];
+    // Validate CLI installation in the background
+    await validateCLI();
```

Comment:
should this be `validateCLI().catch((err:Error)=>...` to avoid blocking activation?